Feature/comments

santiagogustavo opened the pull request:
**Obs:** Revisar o PR de tags antes! (esta branch é uma ramificação)

Sistema de comentários, com as funções de Adicionar, Editar e Deletar.

A listagem é feita na exibição dos posts, cada comentário possui uma célula com as informações:
- Usuário (para essa versão é um valor manualmente inserido, mas com o sistema de usuários passará a ser o usuário que postou);
- Content, com o texto escrito (limite de 128 caracteres);
- Ícones para editar e excluir o comentário, que redirecionam para as rotas corretamente.
Nas telas de adição/edição de comentário, o tem-se um botão para voltar para a página do post (este pega a referência a partir de um parâmetro na URL) e, ao sucesso na operação, retorna para o post atualizado.

Neste PR também adicionei a classe do Font-Awesome e fiz algumas modificações na organização dos arquivos e de alguns métodos.

Commit:
(+) Comment model and controller added, minor changes in code organization

- Edits Makefile to include routing in make all

- Adds Comment model and controller: username and content are required
for validation, Comment belongs to Post model and the comment listing
was added to Post's show route

- Comment creation properly working, with navigation helpers that redirects
from post to comment creation route and from comment creation back to the original
post route

- Changes in Stylesheet class organization, for a better semantical approach

Commit:
(+) Edit and Delete Comment actions implemented, adds icons and handles navigation

- Edit and Delete actions were added and work as expected; seed file updated
to include a Comment instance

- Post now shows a nice message when no comments are present; minor enhancements to
page layout

- Partials moved between views, for a better organization

guilpejon commented on blogster/app/assets/stylesheets/layout.scss:
```diff
@@ -19,12 +19,31 @@ body {
+  font-weight: bold;
+}
+
+.button-margin-5 {
```
Bootstrap tem helpers pra isso: https://v4-alpha.getbootstrap.com/utilities/spacing/

guilpejon commented on blogster/app/controllers/comments_controller.rb:
```diff
@@ -0,0 +1,58 @@
+  end
+
+  def find_post_name
+    @post_name = Post.find(@comment.post_id).title
```
não daria pra fazer `@comment.post.title`?

guilpejon commented on blogster/app/controllers/comments_controller.rb:
```diff
@@ -0,0 +1,58 @@
+  # Obtains the correct Comment with before_action and destroys it
+  def destroy
+    @comment.destroy
+    redirect_to request.referer
```
dá uma olhada no redirect_back https://blog.bigbinary.com/2016/02/29/rails-5-improves-redirect_to_back-with-redirect-back.html

santiagogustavo replied:
Fiquei na dúvida entre o redirect_to e o redirect_back por questões de compatibilidade, mas vou mudar para o segundo :)

guilpejon commented on blogster/app/assets/stylesheets/layout.scss:
```diff
@@ -19,12 +19,31 @@ body {
+}
+
+.button-margin-5 {
+  margin: 5px;
```
Você pode usar spacing utilities do bootstrap pra isso se quiser https://getbootstrap.com/docs/4.0/utilities/spacing/

santiagogustavo replied:
Boa, não me atentei ao detalhe que o Bootstrap tinha essas macros

guilpejon commented on blogster/app/controllers/comments_controller.rb:
```diff
@@ -0,0 +1,58 @@
+  end
+
+  def find_post_name
+    @post_name = Post.find(@comment.post_id).title
```
não daria pra fazer `@comment.post.title`?

santiagogustavo replied:
Sim, fica mais simples!

guilpejon commented on blogster/app/views/comments/buttons/_back.html.erb:
```diff
@@ -0,0 +1 @@
+<%= link_to "Back", post_path(@comment.post_id), class: "btn btn-default" %>
```
você pode usar `<%= link_to "Back", :back, class: "btn btn-default" %>` caso sempre seja para voltar para a página onde ele estava antes

santiagogustavo replied:
Legal, isso vai facilitar bastante!

Commit:
(~) Minor changes and corrections in code, as requested in the PR

- Changes the "Back" button behavior so it handles to the previous page
with the :back value; removes comments/buttons/back and replaces its
references with shared/buttons/back

- Replaces Post.find(@comment.post_id).title with a more elegant statement
@comment.post.title

- Removes the style class button-margin-5 and replaces its references
with Bootstrap's m-5

- Replaces redirect_to request_referrer with redirect_back and a fallback
location

santiagogustavo commented:
@guilpejon corrigi os pontos que vc comentou no código e apliquei um rebase para limpar a lista de commits
